replicator/replicator.go

```diff
@@ -310,14 +322,38 @@ func config_replica(m_ctx context.Context, m_cli immuclient.ImmuClient, r_ctx co
 		}
 	}
 	settings := settings_resp.Settings
+
+	synAcks := settings_resp.Settings.ReplicationSettings.SyncAcks.GetValue()
+	isSynchronousDatabase := synAcks > 0
 	replication_settings := schema.ReplicationNullableSettings{
-		Replica:          &schema.NullableBool{Value: true},
-		MasterDatabase:   &schema.NullableString{Value: src_db},
-		MasterAddress:    &schema.NullableString{Value: config.MasterAddr},
-		MasterPort:       &schema.NullableUint32{Value: uint32(config.MasterPort)},
-		FollowerUsername: &schema.NullableString{Value: config.FollowerUsername},
-		FollowerPassword: &schema.NullableString{Value: config.FollowerPassword},
+		Replica:                      &schema.NullableBool{Value: true},
+		MasterDatabase:               &schema.NullableString{Value: src_db},
+		MasterAddress:                &schema.NullableString{Value: config.MasterAddr},
+		MasterPort:                   &schema.NullableUint32{Value: uint32(config.MasterPort)},
+		FollowerUsername:             &schema.NullableString{Value: config.FollowerUsername},
+		FollowerPassword:             &schema.NullableString{Value: config.FollowerPassword},
+		PrefetchTxBufferSize:         &schema.NullableUint32{Value: uint32(config.PrefetchTxBufferSize)},
+		ReplicationCommitConcurrency: &schema.NullableUint32{Value: uint32(config.ReplicationCommitConcurrency)},
+	}
+
+	switch config.ReplicationSync {
+	case "async": // asynchronous database settings
+		if isSynchronousDatabase { // verifying that the primary database is not setup for synchronous replication
+			return errors.New("not an asynchronous database")
```

Comment:
That will preclude to have an asynchronous replication for a synchronously replicated master.

Imagine this scenario:
master
  - synchronous replica for HA
  - asynchronous replica for backup
  
So while it's correct to deny a sync replica to an async master, it's not correct to deny an async replica to a sync master

Reply:
done, removed the check